src/cargo/core/resolver/mod.rs

```diff
@@ -884,80 +884,205 @@ fn generalize_conflicting(
+                            Some(backtrack_critical_age)
+                        } else {
+                            cx.is_active(id).filter(|&age|
+                                        // we only care about things that are older then critical_age
+                                        age < backtrack_critical_age)
+                        }
+                    },
+                    Some(other.summary.package_id()),
+                ) {
+                    con.extend(
+                        conflicting
+                            .iter()
+                            .filter(|(&id, _)| id != other.summary.package_id())
+                            .map(|(&id, r)| (id, r.clone())),
                     );
+                    println!("used new {}", other.summary.version());
+                    continue;
                 }
-                past_conflicting_activations.insert(dep, &con);
-                return Some(con);
+
+                continue 'dep;
+            }
+
+            if cfg!(debug_assertions) {
+                // the entire point is to find an older conflict, so let's make sure we did
+                let new_age = con
+                    .keys()
+                    .map(|&c| cx.is_active(c).expect("not currently active!?"))
+                    .max()
+                    .unwrap();
+                assert!(
+                    new_age < backtrack_critical_age,
+                    "new_age {} < backtrack_critical_age {}",
+                    new_age,
+                    backtrack_critical_age
+                );
             }
+            past_conflicting_activations.insert(dep, &con);
+            return Some(con);
         }
     }
     None
 }
 
+/// Sees if the candidate is not usable because one of its children will have a conflict
+///
+/// Panics if the input conflict is not all active in `cx`.
+fn generalize_children_conflicting(
+    cx: &Context,
+    critical_age: usize,
+    registry: &mut RegistryQueryer<'_>,
+    past_conflicting_activations: &mut conflict_cache::ConflictCache,
+    dep: &Dependency,
+    candidate: &Summary,
+    parent: PackageId,
+) -> Option<ConflictMap> {
+    let method = Method::Required {
+        dev_deps: false,
+        features: Rc::new(dep.features().iter().cloned().collect()),
+        all_features: false,
+        uses_default_features: dep.uses_default_features(),
+    };
+    let mut out = ConflictMap::new();
+    match registry.build_deps(Some(parent), candidate, &method) {
+        Err(ActivateError::Fatal(_)) => return None,
+        Err(ActivateError::Conflict(pid, reason)) => {
+            out.insert(pid, reason);
+        }
+        Ok(deps) => {
+            let con = deps
+                .1
+                .iter()
+                .filter_map(|(ref new_dep, _, _)| {
+                    past_conflicting_activations.find(
+                        new_dep,
+                        &|id| {
+                            if id == candidate.package_id() {
```

Comment:
This seems pretty similar to the code above, so maybe an implementation could be shared?